EIPS/eip-00.md

```diff
@@ -0,0 +1,790 @@
+This transaction's ID is defined as (`AA_TX_TYPE transaction hash + 1`).
+
+If an AA transaction is included in a block, returns the following values in addition to the existing fields:
+
+| Name                       | Value                                                                        |
+|----------------------------|------------------------------------------------------------------------------|
+| sender                     | Address of the sender of this transaction                                    |
+| nonce                      | The transaction nonce value                                                  |
+| paymaster                  | Address of the Paymaster if it is paying for the transaction, null otherwise |
+| deployer                   | Address of the Deployer if it is included in the transaction, null otherwise    |
+| senderCreationGasUsed      | The amount of gas actually used by the sender deployment frame               |
+| senderValidationGasUsed    | The amount of gas actually used by the sender validation frame               |
+| paymasterValidationGasUsed | The amount of gas actually used by the paymaster validation frame            |
+
+Accepts hash of Transaction Type `AA_TX_TYPE`.
+
+If an AA transaction is included in a block, returns the following values in addition to the existing fields:
+
+| Name                            | Value                                                                                          |
+|---------------------------------|------------------------------------------------------------------------------------------------|
+| status                          | Either 1 (success) or 0 (failure) status of the execution frame                                |
+| executionGasUsed                | The amount of gas actually used by the execution frame                                         |
+| postPaymasterTransactionStatus  | Either 1 (success), 0 (failure), or `null` (did not run) status of the `postPaymasterTransaction` frame |
+| postPaymasterTransactionGasUsed | The amount of gas actually used by the paymaster `postPaymasterTransaction` frame                       |
+
+Note that the field `to` is not included as there is no clear `target` in an `AA_TX_TYPE` transaction.
+
+#### `eth_call`
+
+Accepts Transaction Type `AA_TX_TYPE` with all fields except `from` and `callData` optional.
+
+Returns the return value of [the `sender` execution frame](#sender-execution-frame).
+
+If provided with `paymasterData` and `deployerData` also executes the corresponding frame.
+
+If any of the frames reverts the call returns the revert data of each reverted frame.
+
+#### `eth_estimateGasAccountAbstraction`
+
+Accepts Transaction Type `AA_TX_TYPE` with fields `validationGasLimit`, `paymasterGasLimit`, `callGasLimit` optional.
+
+Returns `{validationGasLimit, paymasterGasLimit, callGasLimit, builderFee}` object.
```

Comment:
One of the most complicated portions of ERC-4337 gas estimation is dealing with ERC-20 (or similar) paymasters that perform a transfer during validation, requiring the user to have enough of the ERC-20 token so that it doesn't revert.

However, a common use case is to first estimate gas, then transfer extra funds for gas, then transact.

This could be overcome by allowing state overrides during gas estimation, where the estimator just overrides the ERC-20 balance of the user to something high for valid gas estimation.

Thoughts on adding state overrides to this gas estimation endpoint?

Reply:
Yes, I think it'll be useful to add state override here.  @forshtat @drortirosh @shahafn any reason not to?

Reply:
I agreed to add stateOverride to estimate.

Reply:
Yes, let's do that.  But also see @StanislavBreadless 's suggestion above, re an automatic stateOverride to seamlessly solve this particular use case?